e2core/satbackend/register.go

```diff
@@ -35,7 +35,7 @@ func registerWithControlPlane(opts options.Options) error {
 		selfIPs = detectedIPs
 	}
 
-	registerURL := fmt.Sprintf("%s/api/v1/upstream/register", opts.ControlPlane)
+	registerURL := fmt.Sprintf("http://%s/api/v1/upstream/register", opts.ControlPlane)
```

Comment:
This appears to imply that the control plane and e2core must be in the same VPC; are we enforcing that now?

Reply:
nice catch!

Reply:
I'd also wrap this inside a `url.Parse`, in case the value of `opts.ControlPlane` is something unexpected. Maybe overkill, but it's a low effort high gain change 🙂

Reply:
Done and done